botocore/session.py

```diff
@@ -796,6 +799,37 @@ def create_client(self, service_name, region_name=None, api_version=None,
             client_config=config, api_version=api_version)
         return client
 
+    def get_available_partitions(self):
+        """Lists the available partitions found on disk
+
+        :rtype: list
+        :return: Returns a list of partition names (e.g., ["aws", "aws-cn"])
+        """
+        resolver = self.get_component('endpoint_resolver')
+        return resolver.get_available_partitions()
+
+    def get_available_regions(self, service_name, partition_name='aws',
```

Comment:
Yeah, that's true. I just pushed a commit that allows you to use the short form of the name or the endpoint_prefix -- e.g., `elb` or `elasticloadbalancing`. We first try to resolve endpoints using the given name. If that result is empty, we then try to load the service data associated with the given service_name to grab it's `endpointPrefix`. If the service cannot be found (e.g., `foo`), then we return an empty list. Otherwise, we return the result of resolving endpoints using the service's `endpointPrefix`.